src/environments/array.js

```diff
@@ -187,12 +194,17 @@ const htmlBuilder: HtmlBuilder<"array"> = function(group, options) {
     let totalHeight = 0;
 
     // Set a position for \hline(s) at the top of the array, if any.
-    for (let i = 1; i <= numHLinesBeforeRow[0]; i++) {
-        if (i >  1) {              // The first \hline doesn't add to height.
-            totalHeight += 0.25;
+    function setHLinePos(hlinesInGap: boolean[]) {
+        let n = 0;
+        while (hlinesInGap.length > 0) {
+            n += 1;
+            if (n > 1) {
+                totalHeight += 0.25;
+            }
+            hlines.push({pos: totalHeight, isDashed: hlinesInGap.pop()});
```

Comment:
Ah, here you are processing in reversed order. I'd find it more intuitive to store them in order, then use a `for` loop or `map` here.